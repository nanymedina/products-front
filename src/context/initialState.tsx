import StateT from '@src/type/state';

const initialState: StateT = {
  products: [],
  search: null,
  shoppingCart: [],
  error: null,
};

export default initialState;
