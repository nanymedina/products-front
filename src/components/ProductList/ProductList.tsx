import { ReactElement } from 'react';
import { Container } from 'react-bootstrap';

import { useStateApp } from '@src/context/appContext';
import ProductT from '@src/type/product';
import Product from './Product';
import Style from './ProductList.module.scss';

const ProductsList = (): ReactElement => {
  const { products } = useStateApp();

  return (
    <Container data-testid="product-list" className={Style.products}>
      {products.map((product: ProductT) => (
        <Product key={product.id} {...product} />
      ))}
    </Container>
  );
};

export default ProductsList;
