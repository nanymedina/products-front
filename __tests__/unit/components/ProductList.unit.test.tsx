import { render } from '@testing-library/react';
import ProductList from '@components/ProductList';
import { useStateApp } from '@src/context/appContext';

jest.mock('@src/context/appContext');

describe('Product List', () => {
  const product = {
    id: 123,
    brand: 'brand',
    description: 'description',
    price: 20,
    oldPrice: null,
    discountRate: null,
    image: 'image',
  };

  const useStateAppMock = useStateApp as jest.Mock;
  useStateAppMock.mockReturnValue({ products: [product, { ...product, id: 456 }] });

  afterEach(() => {
    useStateAppMock.mockClear();
  });

  it('should show product list', () => {
    const { getByTestId, getAllByTestId } = render(<ProductList />);
    const products = getAllByTestId('product');
    const productList = getByTestId('product-list');

    expect(products.length).toEqual(2);
    expect(productList).toMatchSnapshot();
  });
});
