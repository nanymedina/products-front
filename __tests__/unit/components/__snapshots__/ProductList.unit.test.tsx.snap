// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List should show product list 1`] = `
<div
  class="container"
  data-testid="product-list"
>
  <div
    data-testid="product"
  >
    <img
      alt="product-123"
      src="svg/image"
    />
    <div>
      <div>
        <strong>
          brand
        </strong>
        <span>
          description
        </span>
      </div>
      <div>
        <div>
          <span>
            $20
          </span>
        </div>
      </div>
    </div>
    <button
      data-testid="button"
      type="button"
    >
      <strong>
        Agregar
      </strong>
    </button>
  </div>
  <div
    data-testid="product"
  >
    <img
      alt="product-456"
      src="svg/image"
    />
    <div>
      <div>
        <strong>
          brand
        </strong>
        <span>
          description
        </span>
      </div>
      <div>
        <div>
          <span>
            $20
          </span>
        </div>
      </div>
    </div>
    <button
      data-testid="button"
      type="button"
    >
      <strong>
        Agregar
      </strong>
    </button>
  </div>
</div>
`;
